torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -911,7 +910,15 @@ class ShapePropagator {
           [this](Node* node) -> type_vec_t {
             if (auto maybe_tensor_types =
                     gatherTensorTypes<DimensionedTensorType>(node)) {
-              return {broadcast(*maybe_tensor_types, getScalarType(node))};
+              auto first_scalar_type = (*maybe_tensor_types)[0]->scalarType();
+              auto second_scalar_type = typeToScalarType(node->inputs()[1]->type());
+              if (isIntegralType(first_scalar_type) && second_scalar_type.has_value() &&
+                  isFloatingType(*second_scalar_type) )
```

Comment:
We have to give up on setting the output type if `typeToScalarType(node->inputs()[1]->type())` returns an optional. 
```

also nit: format code plz